pkg/id/id_test.go

```diff
@@ -0,0 +1,373 @@
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(tt *testing.T) {
+			tt.Parallel()
+			result, err := NewIDWith(tc.input)
+			exResult, exErr := FromID(tc.input)
+			assert.Equal(tt, exResult, result)
+			assert.Equal(tt, exErr, err)
+		})
+	}
+}
+
+func TestID_FromID(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    string
+		expected struct {
+			result ID
+			err    error
+		}
+	}{
+		{
+			name:  "Fail:Not valid string",
+			input: "testMustFail",
+			expected: struct {
+				result ID
+				err    error
+			}{
+				ID{},
+				ErrInvalidID,
+			},
+		},
+		{
+			name:  "Fail:Not valid string",
+			input: "",
+			expected: struct {
+				result ID
+				err    error
+			}{
+				ID{},
+				ErrInvalidID,
+			},
+		},
+		{
+			name:  "success:valid string",
+			input: "01f2r7kg1fvvffp0gmexgy5hxy",
+			expected: struct {
+				result ID
+				err    error
+			}{
+				ID{ulid.MustParse("01f2r7kg1fvvffp0gmexgy5hxy")},
+				nil,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(tt *testing.T) {
+			tt.Parallel()
+			result, err := FromID(tc.input)
+			assert.Equal(tt, tc.expected.result, result)
+			if err != nil {
+				assert.True(tt, errors.As(tc.expected.err, &err))
+			}
+		})
+	}
+}
+
+func TestID_FromIDRef(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    string
+		expected *ID
+	}{
+		{
+			name:     "Fail:Not valid string",
+			input:    "testMustFail",
+			expected: nil,
+		},
+		{
+			name:     "Fail:Not valid string",
+			input:    "",
+			expected: nil,
+		},
+		{
+			name:     "success:valid string",
+			input:    "01f2r7kg1fvvffp0gmexgy5hxy",
+			expected: &ID{ulid.MustParse("01f2r7kg1fvvffp0gmexgy5hxy")},
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(tt *testing.T) {
+			tt.Parallel()
+			result := FromIDRef(&tc.input)
+			assert.Equal(tt, tc.expected, result)
+			if tc.expected != nil {
+				assert.Equal(tt, *tc.expected, *result)
+			}
+		})
+	}
+}
+
+func TestID_MustBeID(t *testing.T) {
+	testCases := []struct {
+		name        string
+		input       string
+		shouldPanic bool
+		expected    ID
+	}{
+		{
+			name:        "Fail:Not valid string",
+			input:       "testMustFail",
+			shouldPanic: true,
+		},
+		{
+			name:        "Fail:Not valid string",
+			input:       "",
+			shouldPanic: true,
+		},
+		{
+			name:        "success:valid string",
+			input:       "01f2r7kg1fvvffp0gmexgy5hxy",
+			shouldPanic: false,
+			expected:    ID{ulid.MustParse("01f2r7kg1fvvffp0gmexgy5hxy")},
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(tt *testing.T) {
+			tt.Parallel()
+
+			if tc.shouldPanic {
+				assert.Panics(tt, func() { MustBeID(tc.input) })
+				return
+			}
+			result := MustBeID(tc.input)
+			assert.Equal(tt, tc.expected, result)
+		})
+	}
+}
+
+func TestID_Copy(t *testing.T) {
+	id := New()
+
+	id2 := id.Copy()
+
+	assert.Equal(t, id.id, id2.id)
```

Comment:
```suggestion
	assert.Equal(t, id.id, id2.id)
	assert.NotSame(t, id.id, id2.id)
```

`NotSame` is also helpful to test if they are not the same pointer.

Reply:
Thanks I was looking how to assert the pointers are equal or not